Bottom: 6ee9eed7d73380d08962fe1daf65aac31d427238
Top:    3bd8d03183192267eeea1a167c3e2e6629b32712
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-10 16:42:12 -0500

Add Gitlab Authentication

---
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index c3c87d10..4fca566a 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -56,13 +56,22 @@ c['www'] = {
         'console_view': True,
         'grid_view': True,
         },
-    'auth':util.UserPasswordAuth([("georgem","tgbyhn")]),
+    'auth': util.GitLabAuth(os.getenv('GITLAB_URI'),
+                            os.getenv('GITLAB_CLIENT_ID'),
+                            os.getenv('GITLAB_CLIENT_SECRET'),
+                            userInfoProvider=userInfoProvider),
     'authz':util.Authz(
         allowRules=[
             util.AnyControlEndpointMatcher(role="admins"),
             ],
         roleMatchers=[
-            util.RolesFromUsername(roles=['admins'], usernames=["georgem"]),
+            util.RolesFromUsername(
+                roles=['admins'],
+                usernames=[
+                    "george.mccollister",
+                    "joseph.lutz",
+                    "andrew.cooper",
+                ]),
             ],
         ),
     }
